File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    /*Variable Declaration */
    float dollars;

    /*Get Change owed input from user until a non-negative number is given*/
    do
    {
        dollars = get_float("Change owed:");
    }
    while (dollars <= 0);

    /*Converting dollars to the cent and rounding up the number to deal with computer division errors */
    int cent = round(100 * dollars);

    /*Finding how many 25 cents exist in given cent, updating cent by remained */
    int c_25 = floor(cent / 25);
    cent = cent - 25 * c_25;

    /*Finding how many 25 cents exist in remained, updating cent by remained */
    int c_10 = floor(cent / 10);
    cent = cent - 10 * c_10;

    /*Finding how many 25 cents exist in remained, updating cent by remained */
    int c_5 = floor(cent / 5);
    cent = cent - 5 * c_5;

    /*Finding how many 25 cents exist in remained change owned*/
    int c_1 = floor(cent);

    printf("%i\n", c_25 + c_10 + c_5 + c_1);

}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    dollars = get_float("Change owed:")
    if dollars > 0:
        break

# Convert dollars to cent    
cent = int(dollars * 100)

# Calculate how many 25 cents in the given money and calculate remaining cents after 25 cents are removed
c_25 = int(cent / 25)
cent = cent - c_25 * 25

# Calculate how many 10 cents in the given money and calculate remaining cents after 10 cents are removed
c_10 = int(cent / 10)
cent = cent - c_10 * 10

# Calculate how many 5 cents in the given money and calculate remaining cents after 5 cents are removed
c_5 = int(cent / 5)
cent = cent - c_5 * 5

# The remaning are 1 cents
c_1 = int(cent)

# Print the total number of metals
print(c_25 + c_10 + c_5 + c_1)